test/gtest/ucp/test_ucp_tag_probe.cc

```diff
@@ -262,8 +263,14 @@ UCS_TEST_P(test_ucp_tag_probe, limited_probe_size) {
 
     skip_loopback();
 
-    sendbuf.resize(100, '1');
-    recvbuf.resize(100, '0');
+    if (has_transport("tcp")) {
+        /* Increase a message size for TCP transport to be able
```

Comment:
why is it needed?

Reply:
If we have `128 Kbytes` buffer on a receiver side and sending `1000` small messages (`100 bytes`):
`100 * 1000 = 100000 < 128 Kbytes`
So the receiver is able to receive them and return all of them to user from `ucp_tag_probe_nb()`
But if we have large enough messages (say, `10 000 bytes`), the receiver is unable to receive them all w/o enough progressing.

Do you think better to calculate the message size based on TCP receiver's buffer size? It will require touching TCP iface structure from UCP code, because we don't have access to UCT config

Reply:
i see. why do we want to send >128K? in order to have better testing coverage?
maybe increase to 1000 for all transports? (unless valgrind for example)

Reply:
yes, 1000 is a good idea, but since we have `EXPECT_LE(recvd, 128);` check below, we should use `1100` instead or calculate a proper value based on the underlying transport

sending `1000 / ucs::test_time_multiplier()` under valgrind will still fail the test w/ TCP TL

Should we skip the test when running under valgrind? Not sure that this is a good option

Reply:
maybe modify TCP buffer size to smaller size, just for this test, in UCS_TEST_P macro?

Reply:
> maybe modify TCP buffer size to smaller size, just for this test, in UCS_TEST_P macro?

good idea
let's do it through `scope_setenv`, because `UCS_TEST_P` macro can't change UCT config parameters when defining UCP test
Anyway it should be done along with Zcopy changes in #3782 (it defines new config parameter for TCP RX buffer size)